src/ts_catalog/compression_settings.c

```diff
@@ -293,21 +323,64 @@ ts_compression_settings_update(CompressionSettings *settings)
 	FormData_compression_settings *fd = &settings->fd;
 	ScanKeyData scankey[1];
 
-	if (settings->fd.orderby && settings->fd.segmentby)
+	if (settings->fd.orderby && (settings->fd.segmentby || settings->fd.index))
 	{
 		Datum datum;
 		bool isnull;
 
 		ArrayIterator it = array_create_iterator(settings->fd.orderby, 0, NULL);
 		while (array_iterate(it, &datum, &isnull))
 		{
-			if (ts_array_is_member(settings->fd.segmentby, TextDatumGetCString(datum)))
+			if (settings->fd.segmentby &&
+				ts_array_is_member(settings->fd.segmentby, TextDatumGetCString(datum)))
+			{
 				ereport(ERROR,
 						(errcode(ERRCODE_SYNTAX_ERROR),
 						 errmsg("cannot use column \"%s\" for both ordering and segmenting",
 								TextDatumGetCString(datum)),
 						 errhint("Use separate columns for the timescaledb.compress_orderby and"
 								 " timescaledb.compress_segmentby options.")));
+			}
+
+			if (settings->fd.index &&
+				ts_contains_sparse_index_config(settings,
+												TextDatumGetCString(datum),
+												ts_sparse_index_type_names
+													[_SparseIndexTypeEnumBloom]))
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_SYNTAX_ERROR),
+						 errmsg("the orderby column \"%s\" cannot have a bloom sparse index ",
+								TextDatumGetCString(datum)),
+						 errhint("Use separate columns for the timescaledb.compress_orderby and"
+								 " timescaledb.compress_index options.")));
```

Comment:
Not sure I get this. Maybe "For orderby columns, a minmax sparse index is added automatically."?